helm/provider.go

```diff
@@ -81,7 +84,12 @@ func Provider() terraform.ResourceProvider {
 							return
 						}
 					}
-					errs = append(errs, fmt.Errorf("%s must be a valid storage driver", v))
+					diags = append(diags, diag.Diagnostic{
+						Severity: diag.Error,
+						Summary:  "Invalid storage driver: " + v + " used for helm_driver",
+						Detail:   "Helm backend storage driver must be set to one of the following values:" + strings.Join(drivers, ","),
```

Comment:
I would propose that we use `fmt.Sprintf` here for these things and keep it consistent so that it's easy to read. Pretty sure `+` is faster but I would optimize for readability here.